src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -123,9 +123,10 @@ trait Formats
         "startedAt" -> launched.status.startedAt,
         "stagedAt" -> launched.status.stagedAt,
         "ports" -> launched.hostPorts,
-        "version" -> launched.appVersion,
-        "ipAddresses" -> launched.ipAddresses
-      )
+        "version" -> launched.appVersion
+      ) ++ launched.ipAddresses.map { ipAddresses =>
+          Json.obj("ipAddresses" -> ipAddresses)
+        }.getOrElse(Json.obj())
```

Comment:
The output of this complicated block and the simpler version are not the same.

The simple version can result in `"ipAddresses": null` if there's no ip address. The more complicated version will skip the property in that case.

I'll update the PR with an intermediate version using foldLeft.